pkg/operation/botanist/etcdencryption.go

```diff
@@ -0,0 +1,332 @@
+	if err != nil {
+		return err
+	}
+	if !enableEtcdEncryption {
+		b.Logger.Info("support for etcd encryption is offered automatically starting with kubernetes version 1.13")
+		return nil
+	}
+	needToWriteConfig := false
+	exists, ecSeed, forcePlaintext, err := b.readEncryptionConfigurationFromSeed()
+	if err != nil {
+		return err
+	}
+	if !exists {
+		// create new passive EncryptionConfiguration if it does not exist yet
+		// and remember to write this created configuration
+		b.Logger.Info("no EncryptionConfiguration found in seed. Creating new passive configuration.")
+		ecSeed, err = encryptionconfiguration.CreateNewPassiveConfiguration()
+		if err != nil {
+			return err
+		}
+		needToWriteConfig = true
+	} else {
+		exists, ecGarden, err := b.readEncryptionConfigurationFromGarden()
+		if err != nil {
+			return err
+		} else if !exists {
+			return fmt.Errorf("EncryptionConfiguration exists in seed but not in garden")
+		}
+		// if it exists already, it needs to be consistent
+		consistent, err := b.isEncryptionConfigurationConsistent(ecSeed, ecGarden)
+		if (err != nil) || !consistent {
+			return err
+		}
+		if forcePlaintext {
+			// if annotation requires plaintext secrets, thou shalt get plaintext secrets
+			if encryptionconfiguration.IsActive(ecSeed) {
+				// deactivate configuration if it is still active
+				b.Logger.Info("de-activating existing active EncryptionConfiguration. Note: shoot secrets will be decrypted in etcd subsequently")
+				encryptionconfiguration.SetActive(ecSeed, false)
+				needToWriteConfig = true
+			}
+		} else if !encryptionconfiguration.IsActive(ecSeed) {
+			// if it is not active already (aescbc as first provider) then set it to active
+			// and remember to write this created configuration
+			b.Logger.Info("activating existing passive EncryptionConfiguration")
+			encryptionconfiguration.SetActive(ecSeed, true)
+			needToWriteConfig = true
+		}
+	}
+	ecSeedYamlBytes, err := encryptionconfiguration.ToYAML(ecSeed)
+	if err != nil {
+		return err
+	}
+	if needToWriteConfig {
+		b.Logger.Info("writing new/updated EncryptionConfiguration")
+		err = b.writeEncryptionConfiguration(ecSeedYamlBytes)
+		if err != nil {
+			return err
+		}
+	}
+
+	checksum := utils.ComputeSHA256Hex(ecSeedYamlBytes)
+	b.mutex.Lock()
+	b.CheckSums[common.EtcdEncryptionSecretName] = checksum
+	b.mutex.Unlock()
+	// enablement of etcd encryption feature done in helm chart of apiserver deployment
+	return nil
+}
+
+// RewriteShootSecretsIfEncryptionConfigurationChanged rewrites a shoot's secrets if the EncryptionConfiguration has changed
+func (b *Botanist) RewriteShootSecretsIfEncryptionConfigurationChanged() error {
+	b.Logger.Info("starting RewriteShootSecretsIfEncryptionConfigurationChanged")
+	// WARNING:
+	// No explicit checking of whether EncryptionConfiguration is contained in a backup.
+	// Be aware of the risk!
+	//
+	// TODO: Ensure this is also agreed upon by Gardener team
+
+	// TODO: contact Amshuman Rao Karaya
```

Comment:
Why are here TODOs? Can we either remove or resolve them?

Reply:
well, I think it is worth mentioning that there is no check that the encryption key is contained in a backup ... the only second place is now Gardener in addition to the secret in the seed which stores the configuration ... 
I wanted to make everyone prominently aware of this fact.

Reply:
@michael-engler The encryption secret is synced to the Garden cluster after it got created and before it is used, I think this should be good enough. There is no problem in mentioning that in a comment - my question was rather about the TODOs. They should be removed.